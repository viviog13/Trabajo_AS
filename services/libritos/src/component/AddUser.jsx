import React from 'react';

const AddUser = (props) => {
  return (
    <form onSubmit={(event) => props.addUser(event)}>
      <div className="field">
        <input
          name="titulo"
          className="input is-large"
          type="text"
          placeholder="Titulo"
          required
          value={props.titulo}  // nuevo
          onChange={props.handleChange}
        />
      </div>
      <div className="field">
        <input
          name="autor"
          className="input is-large"
          type="text"
          placeholder="Autor"
          required
          value={props.autor}  // nuevo
          onChange={props.handleChange}
        />
      </div>
       <div className="field">
        <input
          name="añodepublicacion"
          className="input is-large"
          type="text"
          placeholder="Año de Publicacion"
          required
          value={props.añodepublicacion}  // nuevo
          onChange={props.handleChange}
        />
      </div>
       <div className="field">
        <input
          name="editorial"
          className="input is-large"
          type="text"
          placeholder="Editorial"
          required
          value={props.editorial}  // nuevo
          onChange={props.handleChange}
        />
      </div>
       <div className="field">
        <input
          name="generoliterario"
          className="input is-large"
          type="text"
          placeholder="Genero literario"
          required
          value={props.generoliterario}  // nuevo
          onChange={props.handleChange}
        />
      </div>
      <input
        type="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
      />
    </form>
  )
};

export default AddUser;