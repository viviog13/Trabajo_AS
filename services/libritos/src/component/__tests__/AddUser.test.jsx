import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddUser from '../AddUser';

test('AddUser renders properly', () => {
  const wrapper = shallow(<AddUser/>);
  const element = wrapper.find('form');
  expect(element.find('input').length).toBe(6);
  expect(element.find('input').get(0).props.name).toBe('titulo');
  expect(element.find('input').get(1).props.name).toBe('autor');
  expect(element.find('input').get(2).props.name).toBe('aÃ±odepublicacion');
  expect(element.find('input').get(3).props.name).toBe('editorial');
  expect(element.find('input').get(4).props.name).toBe('generoliterario');
  expect(element.find('input').get(5).props.type).toBe('submit');
});

test('AddUser renders a snapshot properly', () => {
  const tree = renderer.create(<AddUser/>).toJSON();
  expect(tree).toMatchSnapshot();
});