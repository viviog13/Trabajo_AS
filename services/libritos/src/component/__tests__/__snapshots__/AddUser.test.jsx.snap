// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="titulo"
      onChange={undefined}
      placeholder="Titulo"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="autor"
      onChange={undefined}
      placeholder="Autor"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="añodepublicacion"
      onChange={undefined}
      placeholder="Año de Publicacion"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="editorial"
      onChange={undefined}
      placeholder="Editorial"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="generoliterario"
      onChange={undefined}
      placeholder="Genero literario"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    type="submit"
    value="Submit"
  />
</form>
`;
